ngx_http_vod_hls.c

```diff
@@ -183,6 +183,16 @@ ngx_http_vod_hls_init_encryption_params(
 		if (drm_info->iv_set)
 		{
 			encryption_params->iv = drm_info->iv;
+
+			// check encryption key method must be aes-128, encryption key format can be "identity" or empty and output_iv set to true then return iv
+			// https://datatracker.ietf.org/doc/html/rfc8216#section-5.2
+			if (conf->hls.encryption_method == HLS_ENC_AES_128 && 
+			(conf->hls.m3u8_config.encryption_key_format.len == 0 || ngx_strncmp(conf->hls.m3u8_config.encryption_key_format.data, "identity", conf->hls.m3u8_config.encryption_key_format.len) == 0) && 
+			conf->hls.output_iv)
```

Comment:
I don't see why we need all this complexity - would be better to just replace -
```
encryption_params->return_iv = FALSE;
```
with -
```
encryption_params->return_iv = conf->hls.output_iv;
```
if the user requested an iv - we return an iv.

Reply:
Did you mean changing the line [174](https://github.com/nonth/nginx-vod-module/blob/68eeed9ee7c331336e36e772af7f9618a43c31c4/ngx_http_vod_hls.c#L174)?

Initially, I only thought about returning the IV received from the external server. However, if you think changing the line 174 is sufficient, I can do that.

Reply:
Done 🙂